### PR TITLE
Track stuff

### PR DESCRIPTION
**Pull request by romainGuiet**
Please find here a new class ItemTracker, to track either spots or surfaces! 
It comes with a TrackSurfaceDemo!

### CODE REVIEW

File: src/main/java/ch/epfl/biop/imaris/EasyXT.java [hunk 1]
```diff
@@ -1966,6 +1965,19 @@ public static StatsCreator create(IDataItemPrx item, String statName, Map<Long,
         }
     }
 
+
+    /**
+     *T his class contains methods directly related to Surface and Spots tracking.
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/imaris/EasyXT.java:1970]: Typo, extra space

File: src/main/java/ch/epfl/biop/imaris/ItemTracker.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                } else if (aMethod.equals("AutoregressiveMotion")) {
+                    spots_tracked = EasyXT.Utils.getImarisApp().GetImageProcessing().TrackSpotsAutoregressiveMotion(spots_totrack, aMaximalDistance, aGapSize, aTrackFiltersString);
+                } else if (aMethod.equals("AutoregressiveMotionExpert")) {
+                    spots_tracked = EasyXT.Utils.getImarisApp().GetImageProcessing().TrackSpotsAutoregressiveMotionExpert(spots_totrack, aMaximalDistance, aGapSize, aIntensityWeight, aTrackFiltersString);
+                }
+
+                aItemTracked = spots_tracked;
+
+            } else if (factory.IsSurfaces(aItem)) {
+                System.out.println("it's a surface");
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/imaris/ItemTracker.java:68]: remove debug comment

File: src/main/java/ch/epfl/biop/imaris/ItemTracker.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                } else if (aMethod == "AutoregressiveMotion") {
+                    surfaces_tracked = EasyXT.Utils.getImarisApp().GetImageProcessing().TrackSurfacesAutoregressiveMotion(surfaces_totrack, aMaximalDistance, aGapSize, aTrackFiltersString);
+                } else if (aMethod == "AutoregressiveMotionExpert") {
+                    surfaces_tracked = EasyXT.Utils.getImarisApp().GetImageProcessing().TrackSurfacesAutoregressiveMotionExpert(surfaces_totrack, aMaximalDistance, aGapSize, aIntensityWeight, aTrackFiltersString);
+                }
+
+                aItemTracked = surfaces_tracked;
+            }
+        }else{
+            System.out.println("Please define a Tracking Method, using setMethod(...), or useConnectedComponents , use...");
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/imaris/ItemTracker.java:89]: not logger?

File: src/main/java/ch/epfl/biop/imaris/ItemTracker.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    public static ItemTrackerBuilder Item(ObjectPrx aItem) throws Error {
+        return ItemTrackerBuilder.setItem(aItem);
+    }
+
+    public ObjectPrx track() throws Error {
+
+        IFactoryPrx factory = EasyXT.Utils.getImarisApp().GetFactory();
+        ObjectPrx aItemTracked = null;
+
+        System.out.println(aMethod);
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/imaris/ItemTracker.java:45]: use log.accept()

File: src/main/java/ch/epfl/biop/imaris/ItemTracker.java (hunk 2 above)
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/ItemTracker.java:68]: use log.accept()

File: src/main/java/ch/epfl/biop/imaris/ItemTracker.java (hunk 3 above)
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/ItemTracker.java:89]: use log.accept()

**Review by lacan — CHANGES_REQUESTED** [review]

File: src/main/java/ch/epfl/biop/imaris/ItemTracker.java (hunk 4 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/imaris/ItemTracker.java:45]: removed, was for debug

**Pull request merged**